extensions/wikia/ImageServing/imageServingHelper.class.php

```diff
@@ -9,25 +9,27 @@ class ImageServingHelper {
 
 	private static $hookOnOff = false; // parser hooks are off
 
-	/**
-	 * @param LinksUpdate $linksUpdate
-	 * @return bool return true to continue hooks flow
-	 */
-	public static function onLinksUpdateComplete( $linksUpdate ) {
-		wfProfileIn(__METHOD__);
-		$images = $linksUpdate->getImages();
-		$articleId = $linksUpdate->getTitle()->getArticleID();
+	public static function onArticleEditUpdates( WikiPage $wikiPage, $editInfo, bool $changed ) {
+		$images = $editInfo->output->getImages();
+		$title = $wikiPage->getTitle();
 
-		if(count($images) === 1) {
-			$images = array_keys($images);
-			self::buildIndex( $articleId, $images);
-			wfProfileOut(__METHOD__);
+		// SRE-109: Leave early if this is a talk page (or equivalent, like comment/Wall/Forum)
+		if ( $title->isTalkPage() ) {
 			return true;
 		}
 
-		$article = new Article($linksUpdate->getTitle());
-		self::buildAndGetIndex( $article );
-		wfProfileOut(__METHOD__);
+		$task = \Wikia\Tasks\Tasks\ImageServingTask::newLocalTask();
+		$task->title( $title );
+
+		// SRE-109: Don't trigger a re-parse if the article has only one image or no images at all
+		if ( count( $images ) <= 1 ) {
+			$task->call( 'createIndexFromImages', array_keys( $images ) );
+		} else {
+			$task->call( 'createIndexFromPageContent' );
+		}
+
+		$task->queue();
```

Comment:
The main change here is deferring the ImageServing index update into a background task.